src/Vizkit3dWorld.cpp

```diff
@@ -330,4 +345,34 @@ void Vizkit3dWorld::applyCameraParams() {
     widget->getView(0)->getCamera()->setProjectionMatrixAsPerspective(osg::RadiansToDegrees(fovy), aspectRatio, zNear, zFar);
 }
 
+void Vizkit3dWorld::removeLinksChild(sdf::ElementPtr sdf_model, std::string child_element){
+
+    sdf::ElementPtr linkElem = sdf_model->GetElement("link");
+
+    while(linkElem){
+        if (linkElem->HasElement(child_element)){
```

Comment:
The if/while here is redundant. Nothing will happen if GetElement(child_element) is NULL, so just remove the if